component/module/string/string.go

```diff
@@ -0,0 +1,160 @@
+// Package string defines the module.string component.
+package string
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/flow/tracing"
+	"github.com/grafana/agent/web/api"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "module.string",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the module.string
+// component.
+type Arguments struct {
+	// Content to load for the module.
+	Content rivertypes.Secret `river:"content,attr"`
```

Comment:
@erikbaranowski WDYT about making this a rivertypes.OptionalSecret so the user can specify whether its sensitive or not?

Reply:
I think that makes sense to experiment with.

Reply:
tested and I like it, so it will be a secret or not based on whether or not the content passed in is a secret. We will want to do something about how the UI renders an entire file in the `is_secret = false` case in a future issue.

Reply:
see #3251 for future UI improvement